qiskit/transpiler/passes/scheduling/align_measures.py

```diff
@@ -0,0 +1,127 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Align measurement instructions."""
+
+from collections import defaultdict
+from typing import List
+
+from qiskit.circuit.delay import Delay
+from qiskit.circuit.measure import Measure
+from qiskit.circuit.parameterexpression import ParameterExpression
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler import InstructionDurations
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.exceptions import TranspilerError
+from qiskit.transpiler.passes.scheduling.alap import ALAPSchedule
+
+
+class AlignMeasures(TransformationPass):
+    """Measurement alignment.
+
+    This is control electronics aware optimization pass.
+    """
+
+    def __init__(self, alignment: int, durations: InstructionDurations):
+        super().__init__()
+        self.alignment = alignment
+        self.durations = durations
+
+    def run(self, dag: DAGCircuit):
+        """Run the measurement alignment pass on `dag`.
+
+        Args:
+            dag (DAGCircuit): DAG to be checked.
+
+        Returns:
+            DAGCircuit: DAG with consistent timing and op nodes annotated with duration.
+
+        Raises:
+            TranspilerError: When instruction duration is not provided.
+        """
+        time_unit = self.property_set["time_unit"]
+
+        if len(dag.op_nodes(Delay)) == 0 or len(dag.op_nodes(Measure)) == 0:
+            # delay is only instruction that induce misalignment
+            # other instructions cannot be arbitrarily stretched
```

Comment:
This instruction cannot be accepted by the backend because this already violate chunk size constraint. I was thinking to add chunk size validation for pulse gate later but I can add this check in this PR too.

Reply:
Just added custom gate validation example in this commit https://github.com/Qiskit/qiskit-terra/pull/6673/commits/72821a3881f8bef53262880a301872907c1ccc2e
I'll revert this commit and can move to separate PR if you wish.